src/test/Distributor.t.sol

```diff
@@ -0,0 +1,215 @@
+    MintableERC20 private incomingToken = new MintableERC20("Wrapped Vechain", "WVET");
+    MintableERC20 private tokenToRecover = new MintableERC20("Recover Me", "RME");
+    address private recipient1 = address(1);
+    address private recipient2 = address(2);
+
+    function setUp() public
+    {
+        distributor = new Distributor(incomingToken);
+    }
+
+    function testIncomingToken() public
+    {
+        // assert
+        assertEq(address(distributor.incomingToken()), address(incomingToken));
+    }
+
+    function testSingleCorrectAllocaation() public
+    {
+        // arrange
+        Allocation[] memory daArray = new Allocation[](1);
+        Allocation memory da = Allocation(recipient1, distributor.BASIS_POINTS_MAX());
+
+        daArray[0] = da;
+
+        // act
+        distributor.setAllocations(daArray);
+
+        // assert
+        assertEq(distributor.getAllocationsLength(), 1);
+        assertEq(distributor.getAllocation(0).weight, distributor.BASIS_POINTS_MAX());
+    }   
+
+    function testTwoCorrectAllocation() public
+    {
+        // arrange 
+        Allocation[] memory daArray = new Allocation[](2);
+        Allocation memory da1 = Allocation(recipient1, 8000);
+        Allocation memory da2 = Allocation(recipient2, 2000);
+
+        daArray[0] = da1;
+        daArray[1] = da2;
+
+        // act
+        distributor.setAllocations(daArray);
+
+        // assert
+        assertEq(distributor.getAllocation(0).weight, 8000);
+        assertEq(distributor.getAllocation(1).weight, 2000);
+    }
+
+    function testExpandAllocation() public 
+    {
+        // arrange 
+        Allocation[] memory daArray = new Allocation[](2);
+        Allocation memory da1 = Allocation(recipient1, 6000);
+        Allocation memory da2 = Allocation(recipient2, 4000);
+        daArray[0] = da1;
+        daArray[1] = da2;
+        distributor.setAllocations(daArray);
+
+        Allocation[] memory newDaArray = new Allocation[](5);
+        for (uint256 i = 0; i < 5; ++i)
+        {
+            newDaArray[i] = Allocation(recipient1, distributor.BASIS_POINTS_MAX() / 5);
+        }
+
+        // act
+        distributor.setAllocations(newDaArray);
+
+        // assert
+        assertEq(distributor.getAllocationsLength(), 5);
+    }
+
+    function testShrinkAllocation() public
+    {
+        // arrange
+        Allocation[] memory daArray = new Allocation[](10);
+        for (uint256 i = 0; i < 10; ++i)
+        {
+            daArray[i] = Allocation(recipient1, distributor.BASIS_POINTS_MAX() / 10);
+        }
+        distributor.setAllocations(daArray);
+        
+        Allocation[] memory newDaArray = new Allocation[](1);
+        newDaArray[0] = Allocation(recipient2, distributor.BASIS_POINTS_MAX());
+        
+        // act
+        distributor.setAllocations(newDaArray);
+
+        // assert
+        assertEq(distributor.getAllocationsLength(), 1);
+    }
+
+    function testShrinkAllocationToZero() public
```

Comment:
interesting test case! This is particularly interesting because the constructor does not require the allocations to be set. Do we want to enable shrinking to zero as a way to stop distribution?